jumanji/environments/logic/graph_coloring/generator.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+import chex
+import jax
+from jax import numpy as jnp
+
+
+class Generator(abc.ABC):
+    @property
```

Comment:
What was the reasoning for making this a property?

Reply:
we need to get num_nodes for several operations in the environment, and it was suggested that I make it a generator property rather than generator spec.